pajbot/web/routes/base/login.py

```diff
@@ -166,3 +195,67 @@ def logout():
     @twitch.tokengetter
     def get_twitch_oauth_token():
         return session.get("twitch_token")
+
+    if not spotify:
+        return
```

Comment:
this kind of trickery is too fancy for my liking

You should always register the methods, and bail out with an error page if spotify is not set up.

Reply:
👍